core/drivers/crypto/stm32/stm32_pka.c

```diff
@@ -863,24 +949,20 @@ static bool is_smaller(const struct stm32_pka_bn *a,
 {
 	unsigned int i = MAX(a->size, b->size) + 1;
 
-	do {
+	for (; i != U(1); i--) {
```

Comment:
Prefer with `i` initial value defined in the `for ()` instructions.
Note that `i` shall start at `MAX()` without the increment previously used and stop once reaching 0:
`for (i = MAX(a->size, b->size); i; i--)`